packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -131,18 +137,25 @@ export class WaferMap extends FoundationElement {
             this.dieLabelsSuffix,
             this.maxCharacters
         );
-        this.renderer = new RenderingModule(this.dataManager, this.canvas);
-        this.zoomHandler = new ZoomHandler(
-            this.canvas,
-            this.zoomContainer,
-            this.dataManager,
-            this.renderer,
-            this.canvasSideLength
+
+        const renderer = new RenderingModule(dataManager, this.canvas);
+        this.eventHandler = new EventHandler(
+            this.parseWaferDataToEventData(dataManager, renderer, this)
         );
-        this.zoomHandler.attachZoomBehavior();
-        this.renderer.drawWafer();
+        renderer.drawWafer();
+    }
+
+    private clearCanvas(width: number, height: number): void {
+        const context = this.canvas.getContext('2d')!;
+        context.clearRect(0, 0, width, height);
     }
 
+    private readonly onDieSelected = (die: WaferMapDie): void => {
+        this.dispatchEvent(
+            new CustomEvent('die-selected', { detail: { die } })
+        );
+    };
```

Comment:
Treat this name as a method that does some action instead of how it's used. It is not a property that should be reassigned with a new method like the name `on-` suggests. Also use the built-in [emit helper](https://github.com/ni/nimble/blob/0bb622feed161189d38bb34cd8761c8bc752cf03/packages/nimble-components/src/menu-button/index.ts#L131) 

```suggestion
    private function emitDieSelected(die: WaferMapDie): void {
        this.$emit('die-selected', { detail: { die } });
    };
```

And below the adapter would be:
```ts
        const eventCallbacks: EventCallbacks = {
            dieSelected: (die) => this.emitDieSelected(die)
        };
```

Reply:
applied